tests/ottest/fixtures/blockchain/Restart.cpp

```diff
@@ -0,0 +1,264 @@
+#include <tuple>
+
+#include "ottest/fixtures/blockchain/Restart.hpp"
+#include "ottest/fixtures/common/User.hpp"
+
+namespace ottest
+{
+const User& Restart_fixture::createUser(
+    const std::string& name,
+    const std::string& words)
+{
+    auto [user_alice, success_alice] =
+        CreateClient(opentxs::Options{}, instance++, name, words, address_);
+    EXPECT_TRUE(success_alice);
+
+    return user_alice;
+}
+
+const User& Restart_fixture::createSenderAlice()
+{
+    return createUser(name_alice, words_alice);
+}
+
+const User& Restart_fixture::createReceiverBob()
+{
+    return createUser(name_bob, words_bob);
+}
+
+void Restart_fixture::mineForBothUsers(
+    const User& user_bob,
+    const User& user_alice)
+{
+    Height begin = targetHeight;
+    auto scan_listener_alice = std::make_unique<ScanListener>(*user_alice.api_);
+    auto scan_listener_bob = std::make_unique<ScanListener>(*user_bob.api_);
+
+    targetHeight += blocks_number_;
+
+    std::cout << "Mine for alice\n";
+    // mine coin for Alice
+    MineBlocks(
+        user_alice,
+        begin,
+        blocks_number_,
+        transaction_in_block_,
+        amount_in_transaction_);
+
+    begin += blocks_number_;
+    targetHeight += blocks_number_;
+
+    std::cout << "Mine for bob\n";
+    // mine coin for Bob
+    MineBlocks(
+        user_bob,
+        begin,
+        blocks_number_,
+        transaction_in_block_,
+        amount_in_transaction_);
+
+    begin += blocks_number_;
+    targetHeight += static_cast<int>(MaturationInterval()) + 1;
+
+    auto scan_listener_external_alice_f = scan_listener_alice->get_future(
+        GetHDAccount(user_alice), bca::Subchain::External, targetHeight);
+    auto scan_listener_internal_alice_f = scan_listener_alice->get_future(
+        GetHDAccount(user_alice), bca::Subchain::Internal, targetHeight);
+    auto scan_listener_external_bob_f = scan_listener_bob->get_future(
+        GetHDAccount(user_bob), bca::Subchain::External, targetHeight);
+    auto scan_listener_internal_bob_f = scan_listener_bob->get_future(
+        GetHDAccount(user_bob), bca::Subchain::Internal, targetHeight);
+
+    std::cout << "Mine targetHeight: " << targetHeight << std::endl;
+    // mine MaturationInterval number block with
+    MineBlocks(begin, static_cast<int>(MaturationInterval()) + 1);
+
+    EXPECT_TRUE(scan_listener_alice->wait(scan_listener_external_alice_f));
+    EXPECT_TRUE(scan_listener_alice->wait(scan_listener_internal_alice_f));
+
+    EXPECT_EQ(
+        GetBalance(user_alice),
+        amount_in_transaction_ * blocks_number_ * transaction_in_block_);
+
+    EXPECT_TRUE(scan_listener_bob->wait(scan_listener_external_bob_f));
+    EXPECT_TRUE(scan_listener_bob->wait(scan_listener_internal_bob_f));
+    EXPECT_EQ(
+        GetBalance(user_bob),
+        amount_in_transaction_ * blocks_number_ * transaction_in_block_);
+    EXPECT_EQ(GetTransactions(user_bob).size(), 1);
+    EXPECT_EQ(GetTransactions(user_alice).size(), 1);
```

Comment:
i would get rid of this two EXPECTs or make them more future proof. With this we can only call `mineForBothUsers` once in the lifetime of the user with extra assumption that the user was not involved in any tx before. You could check if `GetTransactions(*).size()` before mine and after mine incremented by one.